src/checkers/electronchecker.py

```diff
@@ -0,0 +1,83 @@
+import logging
+import base64
+from datetime import datetime
+import typing as t
+
+from yarl import URL
+import ruamel.yaml
+
+from ..lib import NETWORK_ERRORS
+from ..lib.externaldata import (
+    Checker,
+    ExternalBase,
+    ExternalData,
+    ExternalFile,
+)
+from ..lib.errors import CheckerQueryError
+from ..lib.checksums import MultiDigest
+from .jsonchecker import parse_timestamp
+
+yaml = ruamel.yaml.YAML(typ="safe")
+log = logging.getLogger(__name__)
+
+
+class ElectronChecker(Checker):
+    CHECKER_DATA_TYPE = "electron-updater"
+    CHECKER_DATA_SCHEMA = {
+        "type": "object",
+        "properties": {
+            "url": {"type": "string", "format": "uri"},
+        },
+    }
+
+    @staticmethod
+    def _read_digests(obj: t.Dict) -> MultiDigest:
+        digests: t.Dict[str, str] = {}
+        for _k in MultiDigest._fields:  # pylint: disable=no-member
+            if _k in obj:
+                digests[_k] = base64.b64decode(obj[_k]).hex()
+        return MultiDigest(**digests)
+
+    async def check(self, external_data: ExternalBase):
+        assert isinstance(external_data, ExternalData)
+
+        if "url" in external_data.checker_data:
+            metadata_url = URL(external_data.checker_data["url"])
+        else:
+            metadata_url = URL(external_data.current_version.url).join(
+                URL("latest-linux.yml")
+            )
+
+        try:
+            async with self.session.get(metadata_url) as resp:
+                metadata = yaml.load(await resp.read())
+        except NETWORK_ERRORS as err:
+            raise CheckerQueryError from err
+
+        if "files" in metadata:
+            # Modern metadata format
+            m_file = metadata["files"][0]
```

Comment:
Is it always the first and only file in this array?

Reply:
Honestly, I'm not sure. [Documentation](https://www.electron.build/auto-update.html) doesn't cover it (or I didn't find it). But from the [electron-updater](https://github.com/electron-userland/electron-builder/tree/master/packages/electron-updater) code, it seems like currently the `file` list always contains only one entry.